File: packages/core/src/scanners/claude-code.ts
import type { ScannerPlugin } from "./scanner-plugin.js";
import type { ScanTarget } from "./index.js";
import { projectTarget, homeTarget } from "./index.js";

export const claudeCodeScanner: ScannerPlugin = {
  agentId: "claude-code",
  agentName: "Claude Code",
  description: "Claude Code agent configuration (prompts, MCP servers, settings, skills)",
  targets(projectPath: string, homeDir: string): ScanTarget[] {
    return [
      projectTarget(projectPath, "CLAUDE.md", "claude-code", "agent_instruction", "markdown"),
      projectTarget(projectPath, ".mcp.json", "claude-code", "agent_config", "json"),
      projectTarget(projectPath, ".claude/settings.json", "claude-code", "agent_config", "json"),
      homeTarget(homeDir, ".claude/settings.json", "claude-code", "agent_config", "json"),
      homeTarget(homeDir, ".claude.json", "claude-code", "agent_config", "json", {
        metadataOnly: true,
        sensitivity: "metadata"
      }),
      homeTarget(homeDir, ".claude/agents", "claude-code", "unsupported", "filesystem", { directory: true }),
      homeTarget(homeDir, ".claude/skills", "claude-code", "skill", "filesystem", { directory: true }),
    ];
  }
};

File: crates/hem-core/src/scan/plugins/claude_code.rs
use std::path::Path;

use crate::types::{AgentId, EvidenceKind, EvidenceParser};

use super::super::{home_target, project_target, ScanTarget, ScanTargetOverrides};
use super::ScannerPlugin;

pub struct ClaudeCodeScanner;

impl ScannerPlugin for ClaudeCodeScanner {
    fn agent_id(&self) -> AgentId {
        AgentId::ClaudeCode
    }

    fn agent_name(&self) -> &'static str {
        "Claude Code"
    }

    fn description(&self) -> &'static str {
        "Claude Code agent configuration (prompts, MCP servers, settings, skills)"
    }

    fn targets(&self, project_path: &Path, home_dir: &Path) -> Vec<ScanTarget> {
        vec![
            project_target(
                project_path,
                "CLAUDE.md",
                AgentId::ClaudeCode,
                EvidenceKind::AgentInstruction,
                EvidenceParser::Markdown,
                ScanTargetOverrides::default(),
            ),
            project_target(
                project_path,
                ".mcp.json",
                AgentId::ClaudeCode,
                EvidenceKind::AgentConfig,
                EvidenceParser::Json,
                ScanTargetOverrides::default(),
            ),
            project_target(
                project_path,
                ".claude/settings.json",
                AgentId::ClaudeCode,
                EvidenceKind::AgentConfig,
                EvidenceParser::Json,
                ScanTargetOverrides::default(),
            ),
            home_target(
                home_dir,
                ".claude/settings.json",
                AgentId::ClaudeCode,
                EvidenceKind::AgentConfig,
                EvidenceParser::Json,
                ScanTargetOverrides::default(),
            ),
            home_target(
                home_dir,
                ".claude.json",
                AgentId::ClaudeCode,
                EvidenceKind::AgentConfig,
                EvidenceParser::Json,
                ScanTargetOverrides {
                    metadata_only: Some(true),
                    sensitivity: Some("metadata".to_string()),
                    ..ScanTargetOverrides::default()
                },
            ),
            home_target(
                home_dir,
                ".claude/agents",
                AgentId::ClaudeCode,
                EvidenceKind::Unsupported,
                EvidenceParser::Filesystem,
                ScanTargetOverrides {
                    directory: Some(true),
                    ..ScanTargetOverrides::default()
                },
            ),
            home_target(
                home_dir,
                ".claude/skills",
                AgentId::ClaudeCode,
                EvidenceKind::Skill,
                EvidenceParser::Filesystem,
                ScanTargetOverrides {
                    directory: Some(true),
                    ..ScanTargetOverrides::default()
                },
            ),
        ]
    }
}

pub fn claude_code_scanner() -> ClaudeCodeScanner {
    ClaudeCodeScanner
}